dtfs-central-api/src/repositories/tfm-deals-repo/tfm-deal-cancellation.repo.ts

```diff
@@ -70,4 +81,30 @@ export class TfmDealCancellationRepo {
 
     return updateResult;
   }
+
+  /**
+   * Deletes the tfm deal cancellation object
+   * @param dealId - The deal id
+   * @param auditDetails - The users audit details
+   */
+  public static async deleteOneDealCancellation(dealId: string | ObjectId, auditDetails: AuditDetails): Promise<void> {
+    if (!ObjectId.isValid(dealId)) {
+      throw new InvalidDealIdError(dealId.toString());
+    }
+
+    const dealCollection = await this.getCollection();
+
+    const updateResult = await dealCollection.updateOne(
+      {
+        _id: { $eq: new ObjectId(dealId) },
+        'tfm.stage': { $ne: TFM_DEAL_STAGE.CANCELLED },
+        'dealSnapshot.submissionType': { $in: [DEAL_SUBMISSION_TYPE.AIN, DEAL_SUBMISSION_TYPE.MIN] },
+      },
+      { $unset: { 'tfm.cancellation': '' }, $set: { auditRecord: generateAuditDatabaseRecordFromAuditDetails(auditDetails) } },
+    );
+
+    if (!updateResult?.matchedCount) {
+      throw new DealNotFoundError(dealId.toString());
+    }
+  }
```

Comment:
Can we/should we return something here? I'd typically expect linting to fail with "expected to return a value" kind of rule

Reply:
What do you have in mind? I dont know if sending the `updateResult` will actually be any use. It could be returned here & sent by the controller, but we're just going to ignore the body anyway in the front end so I dont think its necessary.

Reply:
I was thinking could return `updateResult` like you say - appreciate the front end is going to use this but I think it should return _something_ as we're already doing this with `deleteMany`